mattermost-plugin/server/configuration_test.go

```diff
@@ -1,90 +1,86 @@
 package main
 
 import (
-	"errors"
 	"testing"
 
 	"github.com/mattermost/focalboard/server/model"
+	"github.com/mattermost/focalboard/server/server"
+	"github.com/mattermost/focalboard/server/services/config"
 	"github.com/mattermost/focalboard/server/ws"
+	serverModel "github.com/mattermost/mattermost-server/v6/model"
 	"github.com/mattermost/mattermost-server/v6/plugin/plugintest"
-	"github.com/mattermost/mattermost-server/v6/plugin/plugintest/mock"
+
 	"github.com/stretchr/testify/assert"
-	"github.com/stretchr/testify/require"
 )
 
-var errLoadPluginConfiguration = errors.New("loadPluginConfiguration Error")
-
-func TestConfiguration(t *testing.T) {
-	t.Run("null configuration", func(t *testing.T) {
-		plugin := &Plugin{}
-		assert.NotNil(t, plugin.getConfiguration())
-	})
-
-	t.Run("changing configuration", func(t *testing.T) {
-		plugin := &Plugin{}
+type TestHelper struct {
+	Server *server.Server
+}
 
-		configuration1 := &configuration{EnablePublicSharedBoards: false}
-		plugin.setConfiguration(configuration1)
-		assert.Equal(t, configuration1, plugin.getConfiguration())
+func SetupTestHelper() *TestHelper {
+	th := &TestHelper{}
+	th.Server = newTestServer()
+	return th
+}
 
-		configuration2 := &configuration{EnablePublicSharedBoards: true}
-		plugin.setConfiguration(configuration2)
-		assert.Equal(t, configuration2, plugin.getConfiguration())
-		assert.NotEqual(t, configuration1, plugin.getConfiguration())
-		assert.False(t, plugin.getConfiguration() == configuration1)
-		assert.True(t, plugin.getConfiguration() == configuration2)
+func newTestServer() *server.Server {
+	srv, err := server.New(server.Params{
+		Cfg: &config.Configuration{},
 	})
+	if err != nil {
+		panic(err)
+	}
 
-	t.Run("setting same configuration", func(t *testing.T) {
-		plugin := &Plugin{}
-
-		configuration1 := &configuration{}
-		plugin.setConfiguration(configuration1)
-		assert.Panics(t, func() {
-			plugin.setConfiguration(configuration1)
-		})
-	})
+	return srv
+}
 
-	t.Run("clearing configuration", func(t *testing.T) {
+func TestConfiguration(t *testing.T) {
+	t.Run("null configuration", func(t *testing.T) {
```

Comment:
This lonely `t.Run` inside a `Test` function, looks weird to me. I would just change the name of the test function to something like `TestNullConfiguration` and remove the extra nesting there.